module/VuFind/tests/unit-tests/src/VuFindTest/ILS/Driver/SierraRestTest.php

```diff
@@ -127,4 +133,190 @@ public function testPrefixedBibFormatting()
+            ],
+            'fixture' => 'patron-response-1',
+            'expected' => [
+                'email' => 'pref@email.fi',
+                'firstname' => 'Testaaja',
+                'lastname' => 'Testeri',
+                'id' => '10921',
+                'major' => null,
+                'college' => null,
+                'cat_username' => '1111',
+                'cat_password' => '1212',
+            ],
+        ];
+        yield 'missing some fields' => [
+            'patron' => [
+                'id' => '1111',
+                'cat_username' => '1111',
+                'cat_password' => '1212',
+            ],
+            'fixture' => 'patron-response-2',
+            'expected' => [
+                'email' => null,
+                'firstname' => '',
+                'lastname' => 'Testaaja',
+                'id' => '00001',
+                'major' => null,
+                'college' => null,
+                'cat_username' => '1111',
+                'cat_password' => '1212',
+            ],
+        ];
+    }
+
+    /**
+     * Test patronLogin
+     *
+     * @param array  $patron   User patron
+     * @param string $fixture  Name of the response fixture file
+     * @param array  $expected Expected results
+     *
+     * @return       void
+     * @dataProvider getTestPatronLoginData
+     */
+    public function testPatronLogin(array $patron, string $fixture, array $expected): void
+    {
+        $request = [
+            'authMethod' => 'native',
+            'patronId' => $patron['cat_username'],
+            'patronSecret' => $patron['cat_password'],
+        ];
+        $requestMap = [
+            [
+                ['v6', 'patrons', 'auth'],
+                json_encode($request),
+                'POST',
+                ['response' => 'something'],
+            ],
+            [
+                ['v6', 'patrons', ['response' => 'something']],
+                ['fields' => 'names,emails'],
+                $this->getJsonFixture('sierrarest/' . $fixture . '.json'),
+            ],
+        ];
+        $driver = $this->createDriver(requestMap: $requestMap);
+        $result = $driver->patronLogin($patron['cat_username'], $patron['cat_password']);
+        $this->assertEquals($expected, $result);
+    }
+
+    /**
+     * Create driver
+     *
+     * @param ?Converter $dateConverter  Date converter
+     * @param ?callable  $sessionFactory Session factory function
+     * @param array      $requestMap     Make request responses map
+     *
+     * @return MockObject&SierraRest
+     */
+    protected function createDriver(
+        ?Converter $dateConverter = null,
+        ?callable $sessionFactory = null,
+        array $requestMap = [],
+    ): MockObject&SierraRest {
+        $dateConverter ??= new \VuFind\Date\Converter();
+        $sessionContainer = $this->getMockBuilder(Container::class)->disableOriginalConstructor()->getMock();
+        $sessionFactory ??= fn ($namespace) => $sessionContainer;
```

Comment:
I think you might be able to reduce this down to a one-liner:

```suggestion
        $sessionFactory ??= fn ($namespace) => $this->createMock(Container::class);
```

Reply:
Seems like it worked! :)